portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -19,14 +19,39 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.*;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private ArrayList<String> messages = new ArrayList<String>(
+      Arrays.asList("Wond'rful website", "Valorous to seeth thee", "I too loveth the soundeth of music")
+  );
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Ankita!</h1>");
+    
+    // Convert the arraylist to JSON
+    String json = convertToJson(messages);
+
+    // Send the JSON as the response
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+  /**
+   * Converts an Arraylist<String> instance into a JSON string using manual String concatentation.
+   */
+  private String convertToJson(ArrayList<String> messages) {
```

Comment:
Looks like, you already added dependency for gson, do you want to use that for serializing response. see: https://stackoverflow.com/questions/27893342/how-to-convert-list-to-a-json-object-using-gson

Reply:
Makes sense! Fascinatingly, gson Unicode escapes certain characters, of which the apostrophe is one (https://stackoverflow.com/questions/11361176/can-someone-clarify-gsons-unicode-encoding). The only (easy-ish) solution seems to be to use disableHtmlEscaping(), but that makes it slightly less clean. 

Edited the hard-coded comments for now, but will work on figuring out the disableHtmlEscaping() soon!